actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -198,10 +198,12 @@ def fragment_cache_key(key)
 
     private
 
-      def fragment_name_with_digest(name) #:nodoc:
-        if @virtual_path
+      def fragment_name_with_digest(name, virtual_path: nil) #:nodoc:
+        virtual_path ||= @virtual_path
+        if virtual_path
           names  = Array(name.is_a?(Hash) ? controller.url_for(name).split("://").last : name)
-          digest = Digestor.digest name: @virtual_path, finder: lookup_context, dependencies: view_cache_dependencies
+          dependencies = defined?(view_cache_dependencies) ? view_cache_dependencies : []
```

Comment:
`view_cache_dependencies` is defined as a helper method and that's why this doesn't work. Maybe we should just move all the tests to `controller/caching_test`? @rafaelfranca what do you think?